src/med_bench/utils/utils.py

```diff
@@ -1,8 +1,84 @@
 import numpy as np
 
-import rpy2.robjects as robjects
-import rpy2.robjects.packages as rpackages
-from rpy2.robjects import pandas2ri, numpy2ri
+# import rpy2.robjects.packages as rpackages
+# from rpy2.robjects import pandas2ri, numpy2ri
+
+import subprocess
+import warnings
+
+
+def check_r_dependencies():
+    try:
+        # Check if R is accessible by trying to get its version
+        # result = subprocess.run(["R", "--version"], capture_output=True, text=True, check=True)
+        subprocess.check_output(["R", "--version"])
+        
+        # If the above command fails, it will raise a subprocess.CalledProcessError and won't reach here
+        
+        # Assuming reaching here means R is accessible, now try importing rpy2 packages
+        import rpy2.robjects.packages as rpackages
+        required_packages = ['causalweight', 'mediation', 'stats', 'base', 'grf', 'plmed']
+
+        for package in required_packages:
+            rpackages.importr(package)
+
+        return True  # All checks passed, R and required packages are available
+
+    except:
+        # Handle the case where R is not found or rpy2 is not installed
+        # print("R or required R packages not available")
+        return False
+
+
+def is_r_installed():
+    try:
+        subprocess.check_output(["R", "--version"])
+        return True
+    except:
+        return False
+
+def check_r_package(package_name):
+    try:
+        import rpy2.robjects.packages as rpackages
+        rpackages.importr(package_name)
+        return True
+    except Exception:
+        return False
+    
+
+def r_dependency_required(required_packages):
+    def decorator(func):
+        def wrapper(*args, **kwargs):
+            if not is_r_installed():
+                warnings.warn("R is not installed or not found. Please install R and set it up correctly in your system.")
```

Comment:
Can you make sure to enforce the 80  characters limit ?

Reply:
The 80 characters limit rule enforced in the last commit. Thank you for your feedback.